### ISSUE TITLE
Deep nested side bar title isn't indented.

### ISSUE THREAD

**Issue by soumi-akira** [vuejs/vitepress#563]
### Describe the bug

When side bar has more than 4 depth nested title, indent is broken.

### Reproduction

.vitepress/config.ts

```ts
import { defineConfig, DefaultTheme } from "vitepress";

export default defineConfig({
  title: "Some document",
  description: "Just playing around.",
  themeConfig: {
    sidebar: {
      "/": [
        {
          text: "1",
          children: [
            {
              text: "2",
              children: [
                {
                  text: "3",
                  children: [
                    {
                      text: "4",
                      children: [
                        {
                          text: "5",
                          children: [
                            {
                              text: "6",
                              children: [],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
  },
});
```

this produce side bar like bellow.

![image](https://user-images.githubusercontent.com/40164629/156514098-7b075999-6c8f-45b1-af12-31f365319335.png)

### Expected behavior

Produce correct indent.

![image](https://user-images.githubusercontent.com/40164629/156514709-5ccd013e-c7ec-4a0e-a708-9919357f2125.png)

The image above is made by changing sub title padding to `0.3rem 1.5rem 0.3rem 5rem` for 5 and `0.3rem 1.5rem 0.3rem 6rem` for 6 with Chrome dev tool.

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
    Memory: 16.17 GB / 31.59 GB
  Binaries:
    Node: 16.12.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.1.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 99.0.4844.51
    Edge: Spartan (44.22000.120.0), Chromium (98.0.1108.62)
    Internet Explorer: 11.0.22000.120
  npmPackages:
    vitepress: ^0.22.3 => 0.22.3
```

### Additional context

I know if we use deep nested sub title, the layout is broken in many case, but at least it should be able to choice whether we use it or not I think.
Even if it's not supported officially, if we get it work by adding some script or something, it's welcome😀

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.vuejs.org).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by kiaking** [comment]
Thanks for the feedback! But closing it due to sidebar has no more nested structure in `theme-next` 🙇‍♂️

**Comment by adam-tylr** [comment]
Is a nested sidebar intentionally not supported or is it something that could be implemented in the future?

**Comment by brc-dd** [comment]
@adam-tylr It's already implemented (#851). We didn't document it though.
